server/server.go

```diff
@@ -0,0 +1,123 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/mux"
+	"net/http"
+
+	"github.com/Clever/cron-admin/db"
+)
+
+func byteHandler(handler func(*http.Request) ([]byte, error)) func(http.ResponseWriter, *http.Request) {
+	return func(rw http.ResponseWriter, req *http.Request) {
+		data, err := handler(req)
+		if err != nil {
+			fmt.Printf("Received handler err: %s\n", err)
```

Comment:
we have a couple other patterns that might be useful here. We use `httpwrapper` in a few places (https://github.com/Clever/district-view-service/blob/master/main.go#L89-L92). It would handle the mongo session cloning and also have built in error-handling and logging. Another alternative is the middleware in catapult, which provides similar functionality (https://github.com/Clever/catapult/blob/master/api/router/router.go#L56-L74).

We want to standardize on a model for writing http services, and we haven't yet, but in the meantime I think either of those might be useful.